daphne/src/roles.rs

```diff
@@ -12,61 +12,19 @@ use crate::{
     hpke::HpkeDecrypter,
     messages::{
         constant_time_eq, AggregateContinueReq, AggregateInitializeReq, AggregateResp,
-        AggregateShareReq, AggregateShareResp, BatchSelector, CollectReq, CollectResp, Id, Nonce,
-        Report, ReportShare, TransitionFailure, TransitionVar,
+        AggregateShareReq, AggregateShareResp, BatchParameter, BatchSelector, CollectReq,
+        CollectResp, Id, Nonce, Report, ReportShare, Time, TransitionFailure, TransitionVar,
     },
     DapAbort, DapAggregateShare, DapCollectJob, DapError, DapGlobalConfig, DapHelperState,
     DapHelperTransition, DapLeaderProcessTelemetry, DapLeaderTransition, DapOutputShare,
-    DapRequest, DapResponse, DapTaskConfig, DapVersion,
+    DapQueryConfig, DapRequest, DapResponse, DapTaskConfig, DapVersion,
 };
 use async_trait::async_trait;
 use prio::codec::{Decode, Encode};
 use rand::prelude::*;
 use std::collections::HashMap;
 use url::Url;
 
-macro_rules! check_batch_param {
-    (
-        $task_config:expr,
-        $batch_selector:expr,
-        $agg_param:expr
-    ) => {{
-        let batch_interval = $batch_selector.unwrap_interval();
-        if !batch_interval.is_valid_for($task_config) {
-            return Err(DapAbort::InvalidBatchInterval);
-        }
-
-        // Check that the aggreation parameter is suitable for the given VDAF.
-        if !$task_config.vdaf.is_valid_agg_param(&$agg_param) {
-            // TODO spec: Define this behavior.
-            return Err(DapAbort::UnrecognizedMessage);
-        }
-    }};
-}
-
```

Comment:
Reviewer note: This has been moved to https://github.com/cloudflare/daphne/pull/131/files#diff-6edcf4a9c3adc1ac1e8061cd876a931a9dedf381bdf3ce49d70b34b95a697205R779-R793.